server/hotrod/src/main/java/org/infinispan/server/hotrod/tx/table/GlobalTxTable.java

```diff
@@ -0,0 +1,385 @@
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import javax.transaction.HeuristicMixedException;
+import javax.transaction.HeuristicRollbackException;
+import javax.transaction.RollbackException;
+import javax.transaction.xa.Xid;
+
+import org.infinispan.Cache;
+import org.infinispan.commands.tx.RollbackCommand;
+import org.infinispan.commands.tx.TransactionBoundaryCommand;
+import org.infinispan.commons.api.BasicCache;
+import org.infinispan.commons.api.Lifecycle;
+import org.infinispan.commons.tx.XidImpl;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.factories.GlobalComponentRegistry;
+import org.infinispan.factories.KnownComponentNames;
+import org.infinispan.factories.annotations.ComponentName;
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.factories.annotations.Start;
+import org.infinispan.factories.annotations.Stop;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.functional.FunctionalMap;
+import org.infinispan.functional.impl.FunctionalMapImpl;
+import org.infinispan.functional.impl.ReadWriteMapImpl;
+import org.infinispan.remoting.rpc.RpcManager;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.remoting.transport.impl.VoidResponseCollector;
+import org.infinispan.server.hotrod.logging.Log;
+import org.infinispan.server.hotrod.tx.table.functions.ConditionalMarkAsRollbackFunction;
+import org.infinispan.server.hotrod.tx.table.functions.SetCompletedTransactionFunction;
+import org.infinispan.server.hotrod.tx.table.functions.SetDecisionFunction;
+import org.infinispan.server.hotrod.tx.table.functions.TxFunction;
+import org.infinispan.server.hotrod.tx.table.functions.XidPredicate;
+import org.infinispan.stream.CacheCollectors;
+import org.infinispan.transaction.LockingMode;
+import org.infinispan.transaction.tm.EmbeddedTransaction;
+import org.infinispan.util.ByteString;
+import org.infinispan.util.TimeService;
+import org.infinispan.util.logging.LogFactory;
+
+import net.jcip.annotations.GuardedBy;
+
+/**
+ * //TODO document this!
+ *
+ * @author Pedro Ruivo
+ * @since 9.4
+ */
+@Scope(value = Scopes.GLOBAL)
+public class GlobalTxTable implements Runnable, Lifecycle {
+
+   //TODO think about the possibility of using JGroups RAFT instead of replicated cache?
+
+   private static final Log log = LogFactory.getLog(GlobalTxTable.class, Log.class);
+   private static final boolean trace = log.isTraceEnabled();
+
+   private final Cache<CacheXid, TxState> storage;
```

Comment:
yes and no. it would require some mapping inside with the cache name and the modifications made. Also, the status would have to be in the mappings (I guess).
In any case, I can investigate further before final